src/core/meshcop/dtls.cpp

```diff
@@ -165,35 +219,137 @@ otError Dtls::Start(bool             aClient,
     mbedtls_ssl_set_bio(&mSsl, this, &Dtls::HandleMbedtlsTransmit, HandleMbedtlsReceive, NULL);
     mbedtls_ssl_set_timer_cb(&mSsl, this, &Dtls::HandleMbedtlsSetTimer, HandleMbedtlsGetTimer);
 
-    rval = mbedtls_ssl_set_hs_ecjpake_password(&mSsl, mPsk, mPskLength);
+    if (!mApplicationCoapSecure)
```

Comment:
Check cipher suite instead of `mApplicationCoapSecure`?

Reply:
Ok. I remove the `mApplicationCoapSecure` Flag and check cipher instead.